packages/babel-plugin-transform-async-to-generator/test/fixtures/regression/4943/exec.js

```diff
@@ -0,0 +1,51 @@
+function mandatory(paramName) {
+  throw new Error(`Missing parameter: ${paramName}`);
+}
+
+function shouldNotHappen() {
+  throw new Error("should not happen");
+}
+
+async function foo({ a, b = mandatory("b") } = {}) {
+  return Promise.resolve(b);
+}
+
+async function bar(a = mandatory("a"), b = mandatory("b")) {
+  return Promise.resolve(b);
+}
+
+assert.doesNotThrow(() => {
+  foo().then(() => true).catch(() => true);
+});
+
+assert.doesNotThrow(() => {
+  bar().then(() => true).catch(() => true);
+});
+
+async function fooUsage() {
+  try {
+    await foo().then(shouldNotHappen);
+  } catch (err) {
+    assert.equal(err.message, "Missing parameter: b");
+  }
+};
+
+async function barUsage() {
+  try {
+    await bar().then(shouldNotHappen);
+  } catch (err) {
+    assert.equal(err.message, "Missing parameter: a");
+  }
+}
+
+fooUsage().catch((err) => {
+  if (err instanceof AssertionError) {
+    throw err;
+  }
+});
+
+barUsage().catch((err) => {
+  if (err instanceof AssertionError) {
+    throw err;
+  }
+});
```

Comment:
I believe the runner expects a promise to be returned if it is going to have async operations that need to be waited for. As it is, this will queue up a bunch of async operations, but not wait for them to complete before moving on to a new test. 

I guess the easiest solution might be to wrap this whole test in `return (async function(){ ... })();`?